errdefs/invalid.go

```diff
@@ -0,0 +1,69 @@
+package errdefs
+
+import (
+	"errors"
+	"fmt"
+)
+
+// InvalidInput is an error interface which denotes whether the opration failed due
+// to a the resource not being found.
+type ErrInvalidInput interface {
+	InvalidInput() bool
+	error
+}
+
+type invalidInputError struct {
+	cause error
+}
+
+func (e *invalidInputError) Error() string {
+	return e.cause.Error()
+}
+
+func (e *invalidInputError) InvalidInput() bool {
+	return true
+}
+
+func (e *invalidInputError) Cause() error {
```

Comment:
This blows the stack because it's infinitely recursive AFAICT:
https://play.golang.org/p/n9J-YPfZpcM

Reply:
Oops :)

Reply:
Clearly missing a test there. Will fix.